templates/commands/goldentest/new_test_cli.go

```diff
@@ -0,0 +1,165 @@
+)
+
+type NewTestCommand struct {
+	cli.BaseCommand
+
+	flags NewTestFlags
+}
+
+func (c *NewTestCommand) Desc() string {
+	return "create a new golden test"
+}
+
+func (c *NewTestCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options] <test_name> <location>
+
+The {{ COMMAND }} create a new golden test.
+
+The "<test_name>" is the name of the test.
+
+The "<location>" is the location of the template.
+`
+}
+
+func (c *NewTestCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+	c.flags.Register(set)
+	return set
+}
+
+func (c *NewTestCommand) Run(ctx context.Context, args []string) (rErr error) {
+	logger := logging.FromContext(ctx)
+
+	if err := c.Flags().Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	tempDir, err := os.MkdirTemp("", "abc-new-test-*")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory: %w", err)
+	}
+	defer func() {
+		rErr = errors.Join(rErr, os.RemoveAll(tempDir))
+	}()
+
+	wd, err := c.WorkingDir()
+	if err != nil {
+		return fmt.Errorf("failed to get working directory: %w", err)
+	}
+
+	fs := &common.RealFS{}
+	logger.DebugContext(ctx, "downloading/copying template")
+	_, templateDir, err := templatesource.Download(ctx, &templatesource.DownloadParams{
+		CWD:         wd,
+		FS:          fs,
+		TempDirBase: tempDir,
+		Source:      c.flags.Location,
+		// using localSourceParser instead of remoteGitSourceParser.
+		GitProtocol: "",
+	})
+	if err != nil {
+		return err //nolint:wrapcheck
+	}
+
+	spec, err := specutil.Load(ctx, fs, templateDir, c.flags.Location)
+	if err != nil {
+		return err //nolint:wrapcheck
+	}
+	logger.DebugContext(ctx, "resolving inputs")
+
+	resolvedInputs, err := input.Resolve(ctx, &input.ResolveParams{
+		FS:       fs,
+		Inputs:   c.flags.Inputs,
+		Prompt:   c.flags.Prompt,
+		Prompter: c,
+		Spec:     spec,
+	})
+	if err != nil {
+		return err //nolint:wrapcheck
+	}
+
+	buf, err := marshalTestCase(resolvedInputs)
+	if err != nil {
+		return fmt.Errorf("failed to marhsal test config data: %w", err)
+	}
+
+	testDir := filepath.Join(c.flags.Location, goldenTestDir, c.flags.NewTestName)
+	testConfigFile := filepath.Join(testDir, configName)
+
+	err = fs.MkdirAll(testDir, common.OwnerRWXPerms)
+	if err != nil {
+		return fmt.Errorf("failed creating %s directory to contain test yaml file: %w", testDir, err)
+	}
+	// file overriding is not allowed.
+	fh, err := fs.OpenFile(testConfigFile, os.O_CREATE|os.O_EXCL|os.O_WRONLY, common.OwnerRWPerms)
+	if err != nil {
+		return fmt.Errorf("OpenFile(%q): %w", testConfigFile, err)
+	}
+	defer func() {
+		rErr = errors.Join(rErr, fh.Close())
+	}()
+	if _, err := fh.Write(buf); err != nil {
+		return fmt.Errorf("write(%q): %w", testConfigFile, err)
+	}
+
+	return nil
+}
+
+func marshalTestCase(resolvedInputs map[string]string) ([]byte, error) {
+	testCase := goldentest.Test{
+		Inputs: mapToVarValues(resolvedInputs),
+	}
+	buf, err := yaml.Marshal(testCase)
```

Comment:
As I use yaml marshal here, the output yaml file won't have single or double quote as the existing example showed https://github.com/abcxyz/abc/blob/1c91e857127e78537695ec7a02efe9e1dc8ff2ad/examples/templates/render/for_each_dynamic/testdata/golden/multiple_envs/test.yaml#L5. But it is still a valid yaml file and I verified the `record` and `verify` still works without single quote or double quote. 

I googled it, looks like golang yaml package can't support marshal with quote?

Reply:
That seems fine to me, thanks.